### PR TITLE
Add .map chapter to dynamic task concept

### PR DESCRIPTION
**Pull request by TJaniF**
I had this waiting in my draft folder for a while. 😊  The dynamic task mapping guide was missing a chapter on the .map function. I also added two lines about mapping limitation configuration. :)

### CODE REVIEW

File: learn/dynamic-tasks.md [hunk 1]
```diff
@@ -317,6 +319,57 @@ The add_nums task will have three mapped instances with the following results:
...
+The `.map()` method was added in Airflow 2.4 and allows you to use any Python function to transform an iterable input before a task dynamically maps over it.
+
+You can call `.map()` directly on a task using the TaskFlow API (`my_upstream_task_flow_task().map(mapping_function)`) or on the output object of a traditional operator (`my_upstream_traditional_operator.output.map(mapping_function)`).
+
+The downstream task is dynamically mapped over the object created by the `.map()` method using either `.expand()` for a single keyword argument or `.expand_kwargs()` for list of dictionaries containing sets of keyword arguments.
+
+The code snippet below shows how to use `.map()` to skip specific mapped tasks based on a logical condition.
+
+- `list_strings` is the upstream task returning a list of strings.
+- `skip_strings_starting_with_skip` is a mapping function transforming the list of strings into a list of modified strings and `AirflowSkipException`s. This function will not show up as an Airflow task.
```
Comment (jwitz) [inline learn/dynamic-tasks.md:335]:
```suggestion
-`skip_strings_starting_with_skip` transforms a list of strings into a list of modified strings and `AirflowSkipExceptions`. In this DAG, the function transforms `list_strings` into a new list called `transformed_list`. This function will not appear as an Airflow task.
```

File: learn/dynamic-tasks.md [hunk 2]
```diff
@@ -317,6 +319,57 @@ The add_nums task will have three mapped instances with the following results:
...
+    if len(string) < 4:
+        return string + "!"
+    elif string[:4] == "skip":
+        raise AirflowSkipException(
+            f"Skipping {string}; as I was told!"
+        )
+    else:
+        return string + "!"
+        
+# transforming the output of the first task with the map function
```
Comment (jwitz) [inline learn/dynamic-tasks.md:356]:
```suggestion
# Transforming the output of the first task with the map function.
```

File: learn/dynamic-tasks.md [hunk 3]
```diff
@@ -317,6 +319,57 @@ The add_nums task will have three mapped instances with the following results:
...
+# for non-TaskFlow operators use
```
Comment (jwitz) [inline learn/dynamic-tasks.md:357]:
```suggestion
# For non-TaskFlow operators, use 
```

File: learn/dynamic-tasks.md [hunk 4]
```diff
@@ -317,6 +319,57 @@ The add_nums task will have three mapped instances with the following results:
...
+
+You can call `.map()` directly on a task using the TaskFlow API (`my_upstream_task_flow_task().map(mapping_function)`) or on the output object of a traditional operator (`my_upstream_traditional_operator.output.map(mapping_function)`).
+
+The downstream task is dynamically mapped over the object created by the `.map()` method using either `.expand()` for a single keyword argument or `.expand_kwargs()` for list of dictionaries containing sets of keyword arguments.
+
+The code snippet below shows how to use `.map()` to skip specific mapped tasks based on a logical condition.
+
+- `list_strings` is the upstream task returning a list of strings.
+- `skip_strings_starting_with_skip` is a mapping function transforming the list of strings into a list of modified strings and `AirflowSkipException`s. This function will not show up as an Airflow task.
+- `mapped_printing_function` dynamically maps over the `transformed_list` object created by using the `.map()` method with the mapping function (`skip_strings_starting_with_skip`) as an argument on the upstream task `list_strings`.
```
Comment (jwitz) [inline learn/dynamic-tasks.md:336]:
```suggestion
- `mapped_printing_function` dynamically maps over the `transformed_list` object.
```
By explaining the previous bullet point again within this bullet, it starts to muddy the waters for what specific behavior is being described. Also, does `mapped_printing_function` appear as a task? Is that a meaningful distinction to note?

**Review by jwitz — APPROVED** [review]
Few minor comments but otherwise LGTM!

**Review by kentdanas — APPROVED** [review]
One minor suggestion, but otherwise looks good to me!

File: learn/dynamic-tasks.md [hunk 5]
```diff
@@ -317,6 +319,57 @@ The add_nums task will have three mapped instances with the following results:
...
+        )
+    else:
+        return string + "!"
+        
+# Transforming the output of the first task with the map function.
+# For non-TaskFlow operators, use 
+# my_upstream_traditional_operator.output.map(mapping_function)
+transformed_list = list_strings().map(skip_strings_starting_with_skip)
+
+# the task using dynamic task mapping on the transformed list of strings
```
Comment (TJaniF) [inline learn/dynamic-tasks.md:361]:
```suggestion
# Use dynamic task mapping on the transformed list of strings.
```

**Review by TJaniF — COMMENTED** [review]
Accepted all suggestions and changed the name of the mapped task :)

File: learn/dynamic-tasks.md [hunk 6]
```diff
@@ -317,6 +319,57 @@ The add_nums task will have three mapped instances with the following results:
...
+
+You can call `.map()` directly on a task using the TaskFlow API (`my_upstream_task_flow_task().map(mapping_function)`) or on the output object of a traditional operator (`my_upstream_traditional_operator.output.map(mapping_function)`).
+
+The downstream task is dynamically mapped over the object created by the `.map()` method using either `.expand()` for a single keyword argument or `.expand_kwargs()` for list of dictionaries containing sets of keyword arguments.
+
+The code snippet below shows how to use `.map()` to skip specific mapped tasks based on a logical condition.
+
+- `list_strings` is the upstream task returning a list of strings.
+-`skip_strings_starting_with_skip` transforms a list of strings into a list of modified strings and `AirflowSkipExceptions`. In this DAG, the function transforms `list_strings` into a new list called `transformed_list`. This function will not appear as an Airflow task.
+- `mapped_printing_function` dynamically maps over the `transformed_list` object created by using the `.map()` method with the mapping function (`skip_strings_starting_with_skip`) as an argument on the upstream task `list_strings`.
```
Comment (TJaniF) [inline learn/dynamic-tasks.md]:
```suggestion
- `mapped_printing_task` dynamically maps over the `transformed_list` object created by using the `.map()` method with the mapping function (`skip_strings_starting_with_skip`) as an argument on the upstream task `list_strings`.
```

**Pull request merged**